lib/client/ipam.go

```diff
@@ -243,19 +250,16 @@ func (c ipams) autoAssign(num int, handleID *string, attrs map[string]string, po
 			for _, p := range allPools.Items {
 				// Don't include disabled pools.
 				if !p.Spec.Disabled {
-					pools = append(pools, p.Metadata.CIDR)
+					pools = append(pools, &p.Metadata.CIDR)
 				}
 			}
-		} else {
-			// Use the given pool.
-			pools = []net.IPNet{*pool}
 		}
 
 		// Iterate over pools and assign addresses until we either run out of pools,
 		// or the request has been satisfied.
 		for _, p := range pools {
 			log.Debugf("Assigning from random blocks in pool %s", p.String())
-			newBlock := randomBlockGenerator(p)
+			newBlock := randomBlockGenerator(*p)
```

Comment:
I don't have a strong opinion on this one since the struct is small - I'll make the switch.